File: tools/sms/src/main/java/top/bettercode/summer/tools/sms/aliyun/Reason.java
package top.bettercode.summer.tools.sms.aliyun;

import com.fasterxml.jackson.annotation.JsonProperty;

public class Reason {

  /**
   * 审批未通过的备注信息。
   */
  @JsonProperty("RejectSubInfo")
  private String rejectSubInfo;

  /**
   * 审批未通过的时间，格式为yyyy-MM-dd HH:mm:ss。
   */
  @JsonProperty("RejectDate")
  private String rejectDate;

  /**
   * 审批未通过的原因。
   */
  @JsonProperty("RejectInfo")
  private String rejectInfo;

  public void setRejectSubInfo(String rejectSubInfo) {
    this.rejectSubInfo = rejectSubInfo;
  }

  public String getRejectSubInfo() {
    return rejectSubInfo;
  }

  public void setRejectDate(String rejectDate) {
    this.rejectDate = rejectDate;
  }

  public String getRejectDate() {
    return rejectDate;
  }

  public void setRejectInfo(String rejectInfo) {
    this.rejectInfo = rejectInfo;
  }

  public String getRejectInfo() {
    return rejectInfo;
  }

  @Override
  public String toString() {
    return
        "Reason{" +
            "rejectSubInfo = '" + rejectSubInfo + '\'' +
            ",rejectDate = '" + rejectDate + '\'' +
            ",rejectInfo = '" + rejectInfo + '\'' +
            "}";
  }
}

File: clients/sms/src/main/kotlin/top/bettercode/summer/tools/sms/aliyun/Reason.kt
package top.bettercode.summer.tools.sms.aliyun

import com.fasterxml.jackson.annotation.JsonProperty

class Reason {
    /**
     * 审批未通过的备注信息。
     */
    @JsonProperty("RejectSubInfo")
    var rejectSubInfo: String? = null

    /**
     * 审批未通过的时间，格式为yyyy-MM-dd HH:mm:ss。
     */
    @JsonProperty("RejectDate")
    var rejectDate: String? = null

    /**
     * 审批未通过的原因。
     */
    @JsonProperty("RejectInfo")
    var rejectInfo: String? = null
    override fun toString(): String {
        return "Reason{" +
                "rejectSubInfo = '" + rejectSubInfo + '\'' +
                ",rejectDate = '" + rejectDate + '\'' +
                ",rejectInfo = '" + rejectInfo + '\'' +
                "}"
    }
}